staging/src/k8s.io/client-go/util/watchlist/watch_list.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watchlist
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientfeatures "k8s.io/client-go/features"
+)
+
+// CanUseWatchListForListRequest determines if the given
+// ListOptions can be used to form a watch-list request,
+// which would result in streaming data from the server.
+func CanUseWatchListForListRequest(opts metav1.ListOptions) bool {
+	if !clientfeatures.FeatureGates().Enabled(clientfeatures.WatchListClient) {
+		return false
+	}
+
+	if len(opts.Continue) > 0 {
+		return false
+	}
+	if opts.Limit > 0 {
```

Comment:
Limit == 0 is the default value for client-go, kubectl [applies a limit of 500](https://github.com/kubernetes/kubernetes/blob/b9ec8b3c4643c36491ba94e48e6b957828c850cb/staging/src/k8s.io/kubectl/pkg/cmd/get/get.go#L157)